app-games/src/main/java/com/aptoide/android/aptoidegames/home/NewAppPromotionalView.kt

```diff
@@ -0,0 +1,172 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import cm.aptoide.pt.download_view.presentation.DownloadUiState
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Downloading
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Error
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Install
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Installed
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Installing
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Migrate
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Outdated
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.ReadyToInstall
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Uninstalling
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Waiting
+import cm.aptoide.pt.download_view.presentation.rememberDownloadState
+import cm.aptoide.pt.feature_apps.data.App
+import cm.aptoide.pt.feature_apps.presentation.AppUiState
+import cm.aptoide.pt.feature_apps.presentation.rememberApp
+import cm.aptoide.pt.feature_home.domain.Bundle
+import com.aptoide.android.aptoidegames.R
+import com.aptoide.android.aptoidegames.appview.buildAppViewRoute
+import com.aptoide.android.aptoidegames.drawables.icons.getAppcoinsClearLogo
+import com.aptoide.android.aptoidegames.installer.presentation.AppIconWProgress
+import com.aptoide.android.aptoidegames.installer.presentation.InstallViewShort
+import com.aptoide.android.aptoidegames.installer.presentation.ProgressText
+import com.aptoide.android.aptoidegames.theme.AGTypography
+import com.aptoide.android.aptoidegames.theme.Palette
+
+@Composable
+fun NewAppPromotionalView(
+  bundle: Bundle,
+  navigate: (String) -> Unit
+) {
+  bundle.view?.let {
+    val (uiState, _) = rememberApp(it)
+
+    when (uiState) {
+      is AppUiState.Idle -> NewAppPromotional(
+        app = uiState.app,
+        label = bundle.title,
+        navigate = navigate
+      )
+
+      is AppUiState.Loading -> LoadingView()
+      is AppUiState.NoConnection,
+      is AppUiState.Error -> Unit
+    }
+  }
+}
+
+@Composable
+private fun NewAppPromotional(app: App, label: String, navigate: (String) -> Unit) {
+  val downloadState = rememberDownloadState(app)
+  val isDownloading by remember(downloadState) {
+    mutableStateOf(isDownloading(downloadState))
```

Comment:
Why do you need a mutable state here?
What about simple `val isDownloading = remember(downloadState) { isDownloading(downloadState) }`?